crates/solver/src/auction_preprocessing.rs

```diff
@@ -1,150 +1,57 @@
-    }
-    orders
+/// Converts a token price from the orderbook API `/auction` endpoint to an
+/// native token exchange rate.
+fn to_native_xrate(price: U256) -> BigRational {
+    // Prices returned by the API are already denominated in native token with
+    // 18 decimals. This means, its value corresponds to how much native token
+    // is needed in order to buy 1e18 of the priced token.
+    // Thus, in order to compute an exchange rate from the priced token to the
+    // native token we simply need to compute `price / 1e18`. This results in
+    // an exchange rate such that `x TOKEN * xrate = y ETH`.
+    price.to_big_rational() / &*UNIT
 }
 
-// In case we cannot estimate the price for a token but there is only one pair traded in the batch the accuracy of the price estimate for ranking a solution doesn't matter that much (surplus is not competing with other tokens).
-// Therefore we can deduct a price estimate by looking at the average limit price of the orders on the single token pair
-fn augment_prices(orders: &[LimitOrder], prices: &mut HashMap<H160, BigRational>) {
-    let first_order = match orders.first() {
-        Some(order) => order,
-        None => return,
-    };
-    let init = (
-        (first_order.sell_token, U256::zero()),
-        (first_order.buy_token, U256::zero()),
-    );
-    let only_pair_with_amounts = orders.iter().try_fold(
-        init,
-        |((first_token, first_amount), (second_token, second_amount)), order| {
-            if first_token == order.sell_token && second_token == order.buy_token {
-                // Same pair as previous (same direction)
-                Some((
-                    (order.sell_token, first_amount + order.sell_amount),
-                    (order.buy_token, second_amount + order.buy_amount),
-                ))
-            } else if first_token == order.buy_token && second_token == order.sell_token {
-                // Same pair as previous (opposite direction)
-                Some((
-                    (order.buy_token, first_amount + order.buy_amount),
-                    (order.sell_token, second_amount + order.sell_amount),
-                ))
-            } else {
-                // Don't augment prices unless all orders are on the same pair
-                None
+/// Converts an `Auction` model's price map into an external price map used by
+/// the driver for objective value computation.
+pub fn to_external_prices(
+    prices: BTreeMap<H160, U256>,
+    native_token: H160,
+) -> HashMap<H160, BigRational> {
+    let mut exchange_rates = prices
+        .into_iter()
+        .map(|(token, price)| (token, to_native_xrate(price)))
+        .collect::<HashMap<_, _>>();
+
+    // Ensure there is a price for the native asset and wrapped native token.
+    // While this is not strictly needed since we are returning a mapping of
+    // exchange rates to the native token, certain components in the driver
+    // expect their existence. The value is just 1.0.
```

Comment:
To me this is a bit of a code smell...

That being said, I didn't feel super comfortable trying to fix at all the locations where it happens. I found one place for sure but its hard to know with confidence if I missed anywhere:

https://github.com/gnosis/gp-v2-services/blob/14c739ad2a4772be348f0f293e9a8740abecf966/crates/solver/src/driver.rs#L338-L342

I created an issue to address this #1666.

Reply:
Changed this to return errors on malformed `/auction` prices and proposed #1669 (although, TBH, I'm not 100% happy with the PR).